File: server/api/v1/ai/template.go
package ai

import (
	"github.com/flipped-aurora/gin-vue-admin/server/global"
	"github.com/flipped-aurora/gin-vue-admin/server/model/ai"
	aiRes "github.com/flipped-aurora/gin-vue-admin/server/model/ai/response"
	"github.com/flipped-aurora/gin-vue-admin/server/model/common/request"
	"github.com/flipped-aurora/gin-vue-admin/server/model/common/response"
	"github.com/flipped-aurora/gin-vue-admin/server/utils"
	"github.com/gin-gonic/gin"
	"github.com/spf13/cast"
	"go.uber.org/zap"
)

type TemplateApi struct{}

// DeleteTemplate
// @Tags      Template
// @Summary   删除模板
// @Security  ApiKeyAuth
// @accept    application/json
// @Produce   application/json
// @Param     data  body      wechat.Template            true  "模板ID"
// @Success   200   {object}  response.Response{msg=string}  "删除模板"
// @Router    /template/template [delete]
func (e *TemplateApi) DeleteTemplate(c *gin.Context) {
	var template ai.Template
	err := c.ShouldBindJSON(&template)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	err = utils.Verify(template.BASEMODEL, utils.IdVerify)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	err = templateService.DeleteTemplate(template)
	if err != nil {
		global.GVA_LOG.Error("删除失败!", zap.Error(err))
		response.FailWithMessage("删除失败", c)
		return
	}
	response.OkWithMessage("删除成功", c)
}

// GetTemplate
// @Tags      Template
// @Summary   获取单一模板信息
// @Security  ApiKeyAuth
// @accept    application/json
// @Produce   application/json
// @Param     data  query     wechat.Template                                                true  "模板ID"
// @Success   200   {object}  response.Response{data=exampleRes.TemplateResponse,msg=string}  "获取单一模板信息,返回包括模板详情"
// @Router    /template/template [get]
func (e *TemplateApi) GetTemplate(c *gin.Context) {
	var template ai.Template
	err := c.ShouldBindQuery(&template)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	err = utils.Verify(template.BASEMODEL, utils.IdVerify)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	data, err := templateService.GetTemplate(cast.ToUint64(template.ID))
	if err != nil {
		global.GVA_LOG.Error("获取失败!", zap.Error(err))
		response.FailWithMessage("获取失败", c)
		return
	}
	response.OkWithDetailed(aiRes.TemplateResponse{Template: data}, "获取成功", c)
}

// GetTemplateList
// @Tags      Template
// @Summary   分页获取权限模板列表
// @Security  ApiKeyAuth
// @accept    application/json
// @Produce   application/json
// @Param     data  query     request.PageInfo                                        true  "页码, 每页大小"
// @Success   200   {object}  response.Response{data=response.PageResult,msg=string}  "分页获取权限模板列表,返回包括列表,总数,页码,每页数量"
// @Router    /template/templateList [get]
func (e *TemplateApi) GetTemplateList(c *gin.Context) {
	var pageInfo request.PageInfo
	err := c.ShouldBindQuery(&pageInfo)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	err = utils.Verify(pageInfo, utils.PageInfoVerify)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	templateList, total, err := templateService.GetTemplateList(utils.GetUserAuthorityId(c), pageInfo)
	if err != nil {
		global.GVA_LOG.Error("获取失败!", zap.Error(err))
		response.FailWithMessage("获取失败"+err.Error(), c)
		return
	}
	response.OkWithDetailed(response.PageResult{
		List:     templateList,
		Total:    total,
		Page:     pageInfo.Page,
		PageSize: pageInfo.PageSize,
	}, "获取成功", c)
}

// UpdateTemplate
// @Tags      AITemplate
// @Summary   审核模板
// @Security  ApiKeyAuth
// @accept    application/json
// @Produce   application/json
// @Param     data  body      wechat.AITemplate            true  "模板ID"
// @Success   200   {object}  response.Response{msg=string}  "审核模板"
// @Router    /template/update [post]
func (e *TemplateApi) UpdateTemplate(c *gin.Context) {
	var template ai.Template
	err := c.ShouldBindJSON(&template)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	err = utils.Verify(template.BASEMODEL, utils.IdVerify)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	err = templateService.Update(template)
	if err != nil {
		global.GVA_LOG.Error("更新失败!", zap.Error(err))
		response.FailWithMessage("更新失败", c)
		return
	}
	response.OkWithMessage("更新成功", c)
}

// CloneTemplate
// @Tags      AITemplate
// @Summary   审核模板
// @Security  ApiKeyAuth
// @accept    application/json
// @Produce   application/json
// @Param     data  body      wechat.AITemplate            true  "模板ID"
// @Success   200   {object}  response.Response{msg=string}  "审核模板"
// @Router    /template/clone [post]
func (e *TemplateApi) CloneTemplate(c *gin.Context) {
	var template ai.Template
	err := c.ShouldBindJSON(&template)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	err = utils.Verify(template.BASEMODEL, utils.IdVerify)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	err = templateService.Clone(template)
	if err != nil {
		global.GVA_LOG.Error("更新失败!", zap.Error(err))
		response.FailWithMessage("更新失败", c)
		return
	}
	response.OkWithMessage("更新成功", c)
}

File: web/src/api/template.js
import service from '@/utils/request'


// @Tags TemplateApi
// @Summary 删除模板
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body dbModel.Template true "生成模板"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /template/generate [delete]
export const generateTemplate = (data) => {
    return service({
        url: '/template/generate',
        method: 'post',
    })
}


// @Tags TemplateApi
// @Summary 删除模板
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body dbModel.Template true "删除模板"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /template/template [delete]
export const deleteTemplate = (data) => {
    return service({
        url: '/template/template',
        method: 'delete',
        data
    })
}

// @Tags TemplateApi
// @Summary 克隆模板
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body dbModel.Template true "克隆"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /template/clone [post]
export const cloneTemplate = (data) => {
    return service({
        url: '/template/clone',
        method: 'post',
        data
    })
}


// @Tags TemplateApi
// @Summary 获取单一模板信息
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body dbModel.Template true "获取单一模板信息"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /template/template [get]
export const getTemplate = (params) => {
    return service({
        url: '/template/template',
        method: 'get',
        params
    })
}

// @Tags TemplateApi
// @Summary 获取权限模板列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body modelInterface.PageInfo true "获取权限模板列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /template/templateList [get]
export const getTemplateList = (params) => {
    return service({
        url: '/template/templateList',
        method: 'get',
        params
    })
}


// @Tags TemplateApi
// @Summary 获取权限模板列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body modelInterface.PageInfo true "获取权限模板列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /template/update [get]
export const updateTemplate = (data) => {
    return service({
        url: '/template/update',
        method: 'post',
        data
    })
}
